src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -911,6 +919,181 @@ public static void M(A<T> t)
+    public void M_NoOpt_AggressiveInlining()
+    {
+    }
+}
+";
+
+            Action<ModuleSymbol> validator = module =>
+            {
+                var c = module.GlobalNamespace.GetMember<NamedTypeSymbol>("C");
+                var aggressiveOptNoInliningMethod = c.GetMember<MethodSymbol>("M_AggressiveOpt_NoInlining");
+#if !NET472 // MethodImplAttributes.AggressiveOptimization was introduced in .NET Core 3
+                Assert.Equal(MethodImplAttributes.AggressiveOptimization | MethodImplAttributes.NoInlining, aggressiveOptNoInliningMethod.ImplementationAttributes);
+#else
+                Assert.Equal((MethodImplAttributes)512 | MethodImplAttributes.NoInlining, aggressiveOptNoInliningMethod.ImplementationAttributes);
+#endif
+
+                var noOptNoInliningMethod = c.GetMember<MethodSymbol>("M_NoOpt_NoInlining");
+                Assert.Equal(MethodImplAttributes.NoOptimization | MethodImplAttributes.NoInlining, noOptNoInliningMethod.ImplementationAttributes);
+
+                var aggressiveOptAggressiveInliningMethod = c.GetMember<MethodSymbol>("M_AggressiveOpt_AggressiveInlining");
+#if !NET472
+                Assert.Equal(MethodImplAttributes.AggressiveOptimization | MethodImplAttributes.AggressiveInlining, aggressiveOptAggressiveInliningMethod.ImplementationAttributes);
+#else
+                Assert.Equal((MethodImplAttributes)512 | MethodImplAttributes.AggressiveInlining, aggressiveOptAggressiveInliningMethod.ImplementationAttributes);
+#endif
+
+                var noOptAggressiveInliningMethod = c.GetMember<MethodSymbol>("M_NoOpt_AggressiveInlining");
+                Assert.Equal(MethodImplAttributes.NoOptimization | MethodImplAttributes.AggressiveInlining, noOptAggressiveInliningMethod.ImplementationAttributes);
+            };
+
+            CompileAndVerify(src, sourceSymbolValidator: validator, symbolValidator: validator);
+        }
+
+        [Fact]
+        public void TestImplementationAttributes_PreserveSigAndRuntime()
+        {
+            var src = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    [MethodImpl(MethodImplOptions.PreserveSig, MethodCodeType = MethodCodeType.Runtime)]
+    public void M()
+    {
+    }
+}
+";
+            Action<ModuleSymbol> validator = module =>
+            {
+                var c = module.GlobalNamespace.GetMember<NamedTypeSymbol>("C");
+                var method = c.GetMember<MethodSymbol>("M");
+                Assert.Equal(MethodImplAttributes.PreserveSig | MethodImplAttributes.Runtime, method.ImplementationAttributes);
+            };
+
+            CompileAndVerify(src, sourceSymbolValidator: validator, symbolValidator: validator);
+        }
+
+        [Fact]
+        public void TestImplementationAttributes_Native()
+        {
+            var src = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    [MethodImpl(MethodCodeType = MethodCodeType.Native)]
+    public void M()
+    {
```

Comment:
@Youssef1313 PEVerify probably doesn't like that you've declared a method with a managed body and marked it `Native`. Probably should make it extern.

Reply:
Looking at another test in our codebase, you can just go ahead and skip PEVerify. Use `verify: Verification.Skipped` in the `CompileAndVerify` options.